azure-cli/keyvault/recover_KeyVaultSecret.ps1

```diff
@@ -0,0 +1,42 @@
+function Recover-KeyVaultSecret {
+  param (
+    [Parameter(Mandatory = $true)]
+    [string]
+    $keyVaultName,
+
+    [Parameter(Mandatory = $true)]
+    [string]
+    $secretName
+  )
+
+  Write-Host "    Recovering $secretName secret"
+  
+  $output = az keyvault secret recover `
+    --vault-name $keyVaultName `
+    --name $secretName
+
+  Throw-WhenError -output $output
+
+  while ($true) {
+    $err = $( $output = az keyvault secret show `
+        --name $secretName `
+        --vault-name $keyVaultName `
+        --query value `
+        --output tsv ) 2>&1
```

Comment:
What does this do?

Reply:
This is a trick to both capture both normal and error output in separate variables such that we can handle the different error/success cases.